ehitchhiking-ui/src/app/shared/components/trips-modal/trips-modal.component.ts

```diff
@@ -56,4 +71,28 @@ export class TripsModalComponent implements OnInit {
 	trackById(index, trip) {
 		return trip.id;
 	}
+
+  filterByRole(){
+	  this.role.isEnable = true;
+	  console.log(this.role.isEnable);
+  }
+
+  changedFavorite(){
+	  console.log('selectedFavorite' + " "+ this.selectedSortByRating);
+  }
+
+  ChangeSort(){
+    if (this.selectedBySort === 0 ){
```

Comment:
pls create a shared enum for sorting states

Reply:
+ use switch

Reply:
Why don't you use SortState  ?

Reply:
used SortState and added switch